core/src/bank_forks.rs

```diff
@@ -165,7 +166,7 @@ impl BankForks {
             for slot in diff.iter() {
                 if **slot > root {
                     let _ = self.add_snapshot(**slot, root);
```

Comment:
Hmmmm why do we want snapshots of non-descendants of root? Shouldn't we be taking snapshots of the parents of root?

Reply:
The snapshots get created for all frozen banks, it is not tracked based on whether they are descendants of root or not. If there are room for optimizations we can look into it.